modules/sdk-coin-atom/src/lib/utils.ts

```diff
@@ -82,14 +82,50 @@ export class Utils implements BaseUtils {
     return hash.match(/^[a-zA-Z0-9]+$/) !== null;
   }
 
+  /**
+   * Checks if a cosmos like Bech32 address matches given regular expression and
+   * validates memoId if present
+   * @param {string} address
+   * @param {RegExp} regExp Regular expression to validate the root address against after trimming the memoId
+   * @returns {boolean} true if address is valid
+   */
+  isValidCosmosLikeAddressWithMemoId(address: string, regExp: RegExp): boolean {
+    if (typeof address !== 'string') return false;
+    const addressArray = address.split('?memoId=');
+    if (
+      ![1, 2].includes(addressArray.length) || // should have at most one occurrence of 'memoId='
```

Comment:
any possibility that `address?memoId=0` is considered as `address` only in UI ?

Reply:
Not sure, since that parsing is implemented in UI, and this function is only used for validation